header.go

```diff
@@ -73,11 +77,12 @@ func (r *Reader) ReadHeader() error {
 			return err
 		}
 
+		// TODO: DRY this out along with other COMPRESSION
```

Comment:
Leaving a comment so you don't forget this (I think it's probably fine to just leave the duplication)